--- /tmp/hgweb_mod.py	2009-05-16 07:27:29.000000000 +0000
+++ hgweb_mod.py	2009-05-16 07:27:45.000000000 +0000
@@ -107,7 +107,7 @@ class hgweb(object):
         # work with CGI variables to create coherent structure
         # use SCRIPT_NAME, PATH_INFO and QUERY_STRING as well as our REPO_NAME
 
-        req.url = req.env['SCRIPT_NAME']
+	req.url = os.path.split(req.env['SCRIPT_NAME'])[0]
         if not req.url.endswith('/'):
             req.url += '/'
         if 'REPO_NAME' in req.env:
