diff -Nupra uncrustify-0.44.old/src/indent.cpp uncrustify-0.44/src/indent.cpp
--- uncrustify-0.44.old/src/indent.cpp	2008-02-26 03:09:19.000000000 +0200
+++ uncrustify-0.44/src/indent.cpp	2008-02-26 03:13:49.000000000 +0200
@@ -682,9 +682,12 @@ void indent_text(void)
                frm.pse[frm.pse_tos].indent     = indent_column + indent_size;
                frm.pse[frm.pse_tos].indent_tmp = frm.pse[frm.pse_tos].indent;
             }
-            else if ((pc->parent_type == CT_CLASS) && !cpd.settings[UO_indent_class].b)
+            else if ((pc->parent_type == CT_CLASS))
             {
-               frm.pse[frm.pse_tos].indent -= indent_size;
+	        if (cpd.settings[UO_indent_class].b)
+	           frm.pse[frm.pse_tos].indent += indent_size;
+	        else
+	           frm.pse[frm.pse_tos].indent -= indent_size;
             }
             else if ((pc->parent_type == CT_NAMESPACE) && !cpd.settings[UO_indent_namespace].b)
             {
@@ -794,16 +797,8 @@ void indent_text(void)
       }
       else if (pc->type == CT_PRIVATE)
       {
-         /* Labels get sent to the left or backed up */
-         if (cpd.settings[UO_indent_access_spec].n > 0)
-         {
-            indent_column = cpd.settings[UO_indent_access_spec].n;
-         }
-         else
-         {
             indent_column = frm.pse[frm.pse_tos].indent +
                             cpd.settings[UO_indent_access_spec].n;
-         }
       }
       else if (pc->type == CT_CLASS_COLON)
       {
